Add a checklist of administrator responsibilities

inoxx03 opened the pull request:
@stoobie @cbudz @bergerhoffer @redhat-documentation/ccs-style-council Please feel free to review the administrator responsibilities and let me know if you'd like me add or change anything. Thank you.

inoxx03 commented:
@stoobie Thank you for the feedback on the draft, I have incorporated your suggestions into this PR. please feel free to let me know if you'd like me to make other changes.

inoxx03 commented:
@cbudz @redhat-documentation/ccs-style-council Please let me know your thoughts  and suggestions about the admin responsibility list. If you think tit is OK, please feel free to approve this PR.

Thank you.

inoxx03 commented:
Thank you for the edits @leswilliams44 :)

stoobie commented on ADMIN.adoc:
```diff
@@ -0,0 +1,35 @@
+* Close issues that are resolved.
+
+=== Driving completion of link:https://github.com/redhat-documentation/supplementary-style-guide/pulls[pull requests]
+
```
* Review open link:https://github.com/redhat-documentation/supplementary-style-guide/pulls[pull requests].

stoobie commented on ADMIN.adoc:
```diff
@@ -0,0 +1,35 @@
+
+== Administrator responsibilities:
+
+As the administrator, you are in charge of performing the following duties:
```
```suggestion
As the administrator, you are the point of contact if any contributors experience issues with the repository.

In addition, you are in charge of the following:
* Maintaining the Style Guide site on GitHub
* Driving resolution of open issues
* Driving completion of pull requests
```

JoanneOFlynn2018 replied:
Instead of "In addition, you are in charge of the following:" in @stoobie's suggested change, how about "You also have the following responsibilities:"

stoobie commented on ADMIN.adoc:
```diff
@@ -0,0 +1,35 @@
+* Ensure that the link:https://redhat-documentation.github.io/supplementary-style-guide/[style guide site] is available and displayed correctly.
+* Create pull requests to update the link:https://github.com/redhat-documentation/supplementary-style-guide/blob/master/README.md[README.md], link:https://github.com/redhat-documentation/supplementary-style-guide/blob/master/GUIDELINES.adoc[GUIDELINES.adoc] and the link:https://github.com/redhat-documentation/supplementary-style-guide/blob/master/CONTRIBUTING.md[Contribution guide], if necessary.
+
+=== Driving resolution of link:https://github.com/redhat-documentation/supplementary-style-guide/issues[open issues]
```
I don't think there should be links in headings. You already have this link in the first bullet.

In general, I think that having duplicate links in the same topic is not user-friendly.

JoanneOFlynn2018 replied:
+1

inoxx03 replied:
@stoobie @JoanneOFlynn2018 Got it, I moved the links form headings to regular text and I removed the duplicates.

stoobie reviewed (dismissed):
@inoxx03 Looks good overall! I just had a few pedantic comments.

Thank you so much for putting together this PR!

ndeevy commented on ADMIN.adoc:
```diff
@@ -0,0 +1,35 @@
+
+As the administrator, you are in charge of performing the following duties:
+
+=== Being the Point of Contact for general issues related to the Supplementary Style Guide repository
```
Maybe point of contact should be lowercase?

cbudz replied:
+1

inoxx03 replied:
It should:) Fixed. Thank you:)

ndeevy commented:
> @stoobie @cbudz @bergerhoffer @redhat-documentation/ccs-style-council Please feel free to review the administrator responsibilities and let me know if you'd like me add or change anything. Thank you.

LGTM!  I left one small comment :)

bergerhoffer commented on ADMIN.adoc:
```diff
@@ -0,0 +1,35 @@
+
+* Review link:https://github.com/redhat-documentation/supplementary-style-guide/issues[issues] submitted to the Supplementary Style Guide GitHub repository.
+* Ensure that discussion on open issues is moving forward consistently. If the discussion does not progress, notify link:https://github.com/orgs/redhat-documentation/teams/ccs-style-council[CCS Style Council] members and report any issues that need your peer's attention at the next meeting.
+* Update issue priority labels according to decisions made in meetings.
```
I would probably remove this, I think this might confuse people more than help. We don't really have priority labels (like low, medium, high). The only thing we have really is the "on hold", which I don't think we need admin rotation members to worry about adding.

cbudz replied:
+1.

inoxx03 replied:
@bergerhoffer That's reasonable. Thank you. I removed the line per your suggestion:)

bergerhoffer reviewed (commented):
@inoxx03 Thanks a lot for drafting these responsibilities! Just a few minor comments.

What do you think about including a section to this page from the README page? Otherwise people might not know that it exists.

JoanneOFlynn2018 commented on ADMIN.adoc:
```diff
@@ -0,0 +1,35 @@
+
+=== Being the Point of Contact for general issues related to the Supplementary Style Guide repository
+
+* You are the point of contact in case other contributors experience issues with the repository.
```
```suggestion
* You are the point of contact if contributors experience issues with the repository.
```

JoanneOFlynn2018 replied:
Being pedantic now rather than someone pointing out these minor issues down the line.

cbudz commented on ADMIN.adoc:
```diff
@@ -0,0 +1,35 @@
+* You are the point of contact in case other contributors experience issues with the repository.
+When people reach out to you about an issue, try to fix it. If you cannot fix the issue, report it to a member of the initiative who can.
+
+=== Maintaining the Style Guide site on GitHub
```
I would also recommend updating the instruction to flag the issue with "Needs Fix" (we can add that if people agree) and simply @ the admin group (redhat-documentation/ccs-style-council)

inoxx03 replied:
> @inoxx03 Thanks a lot for drafting these responsibilities! Just a few minor comments.
> 
> What do you think about including a section to this page from the README page? Otherwise people might not know that it exists.

@bergerhoffer I think that is a good point. Let me add the link after I incorporate the collective feedback form project members. Thank you.

cbudz commented on ADMIN.adoc:
```diff
@@ -0,0 +1,35 @@
+* Ensure that work on open pull requests is moving forward consistently.
+* Ensure that at least three members of the link:https://github.com/orgs/redhat-documentation/teams/ccs-style-council[CCS Style Council] review and approve every contribution before merging it. You can be one of the three reviewers.
+* If a contribution is not reviewed or approved for longer than 1 week, reach out to individual Style Council members and ask them to review and approve the contribution.
+* Merge pull requests with contributions that are approved by three or more Style Council members.
```
@bergerhoffer @inoxx03 Style Council is also just an informal name that doesn't have any defined meaning outside of CCS, so maybe just referring to "project members" is best across this doc.

inoxx03 replied:
@cbudz @bergerhoffer Sounds good. Replaced references to _CCS style council_ with the more generically-worded term _project members_.

inoxx03 commented on ADMIN.adoc:
```diff
@@ -0,0 +1,35 @@
+=== Maintaining the Style Guide site on GitHub
+
+* Ensure that the link:https://redhat-documentation.github.io/supplementary-style-guide/[style guide site] is available and displayed correctly.
+* Create pull requests to update the link:https://github.com/redhat-documentation/supplementary-style-guide/blob/master/README.md[README.md], link:https://github.com/redhat-documentation/supplementary-style-guide/blob/master/GUIDELINES.adoc[GUIDELINES.adoc] and the link:https://github.com/redhat-documentation/supplementary-style-guide/blob/master/CONTRIBUTING.md[Contribution guide], if necessary.
```
Good point! Thank you. Changed

inoxx03 commented on ADMIN.adoc:
```diff
@@ -0,0 +1,36 @@
+=== Driving resolution of open issues
+
+* Review link:https://github.com/redhat-documentation/supplementary-style-guide/issues[issues] submitted to the supplementary style guide GitHub repository.
+* If none of the project members respond to a newly opened issue, add the _Needs Fix_ label to the issue and mention @redhat-documentation/ccs-style-council in a comment on the issue to notify project members about it.
```
> I would also recommend updating the instruction to flag the issue with "Needs Fix" (we can add that if people agree) and simply @ the admin group (redhat-documentation/ccs-style-council)

@cbudz I added a note about the _Needs Fix_ label, but was not sure where you wanted me to place it. Is it OK if I include it under _Driving resolution of open issues_?

stoobie replied:
@inoxx03 That does look like a logical place to me.

cbudz replied:
@inoxx03 that works for me.

inoxx03 commented:
@bergerhoffer @cbudz @stoobie @JoanneOFlynn2018 @ndeevy Thank you for the feedback. I addressed your comments and updated the list. Please let me know what you think.

inoxx03 commented on README.md:
```diff
@@ -18,6 +18,10 @@ If you have a suggestion or question, open an [issue](https://github.com/redhat-
 
+## Administrator responsibilities
+
+See the list of [administrator responsibilities](ADMIN.adoc) for more information on what to do in your role as the administrator of this repository.  
```
@bergerhoffer I linked to the admin responsibilities list from README.adoc. Please let me know whether this is OK, or you'd like me to change anything about this section. Thank you.

ndeevy commented on README.md:
```diff
@@ -18,6 +18,10 @@ If you have a suggestion or question, open an [issue](https://github.com/redhat-
 
 See the [Contributing guide](CONTRIBUTING.md) for information on how to contribute an update.
```
Such a tiny thing but it should be 'information about' not 'information on' to be fully IBM-style compliant. Same for the Administrator responsibilities below.
Overall looks great to me though! Thanks @inoxx03

inoxx03 replied:
@ndeevy Very true:) Thank you. Although, I don't think that I added or changed this line in my current PR, let me fix it anyway:)

inoxx03 replied:
```suggestion
See the [Contributing guide](CONTRIBUTING.md) for information about how to contribute an update.
```

cbudz commented on ADMIN.adoc:
```diff
@@ -0,0 +1,36 @@
+Every 2 weeks a different project member assumes the role of repository administrator.
+Signing up to be an administrator is voluntary.
+To volunteer to be an administrator, contact @cbudz or @bergerhoffer.
+
```
@inoxx03 please insert at the end of this section that each administration rotation is marked on the Style Guide shared calendar, and admins will receive a notification and calendar invite when it's their turn.

cbudz commented on ADMIN.adoc:
```diff
@@ -0,0 +1,36 @@
+* Review link:https://github.com/redhat-documentation/supplementary-style-guide/issues[issues] submitted to the supplementary style guide GitHub repository.
+* If none of the project members respond to a newly opened issue, add the _Needs Fix_ label to the issue and mention @redhat-documentation/ccs-style-council in a comment on the issue to notify project members about it.
+//* Ensure that discussion on open issues is moving forward consistently. If the discussion does not progress, notify link:https://github.com/orgs/redhat-documentation/teams/ccs-style-council[project council members] about the issue.
+* Close issues that are resolved.
```
suggestion: "include a link to the PR that resolves the issue, if one exists."

cbudz reviewed (commented):
Added a few comments/suggestions here @inoxx03

bergerhoffer commented on ADMIN.adoc:
```diff
@@ -0,0 +1,36 @@
+
+Every 2 weeks a different project member assumes the role of repository administrator.
+Signing up to be an administrator is voluntary.
+To volunteer to be an administrator, contact @cbudz or @bergerhoffer.
```
Would you mind making these links to https://github.com/cbudz and https://github.com/bergerhoffer?

bergerhoffer commented on ADMIN.adoc:
```diff
@@ -0,0 +1,36 @@
+=== Driving resolution of open issues
+
+* Review link:https://github.com/redhat-documentation/supplementary-style-guide/issues[issues] submitted to the supplementary style guide GitHub repository.
+* If none of the project members respond to a newly opened issue, add the _Needs Fix_ label to the issue and mention @redhat-documentation/ccs-style-council in a comment on the issue to notify project members about it.
```
Maybe make @redhat-documentation/ccs-style-council a link to https://github.com/orgs/redhat-documentation/teams/ccs-style-council?

bergerhoffer replied:
Also, what is the "Needs fix" label for? I see that we don't have this label and I'm not sure I understand its purpose here. CC @cbudz

bergerhoffer replied:
After discussing, the Needs fix label isn't needed. CC @inoxx03

inoxx03 replied:
Thank you @bergerhoffer. I'll update the PR and remove the line about the _Needs Fix_ label.

bergerhoffer reviewed (commented):
Appreciate the updates, just a few more comments.

bergerhoffer commented:
@inoxx03 Can you take a look at these last few comments? Then I think this should be ready to go!

inoxx03 commented:
@cbudz @bergerhoffer I've updated the PR and added information to the admin responsibilities checklist based on your suggestions. Please let me know if you'd like me to make any further changes. if the list looks OK to you now, please go ahead and merge this PR. Thank you:)

gregraka reviewed (dismissed):
LGTM

bergerhoffer commented on ADMIN.adoc:
```diff
@@ -0,0 +1,37 @@
+
+Every 2 weeks a different project member assumes the role of repository administrator.
+Signing up to be an administrator is voluntary.
+To volunteer to be an administrator, contact link:https://github.com/cbudz[@cbudz] or link:https://github.com/cbudz[@bergerhoffer].
```
The second link needs to be updated to be bergerhoffer

gregraka replied:
@inoxx03 Would you be able to push a new commit with Andrea's requested change (earlier)?

Thanks, --Greg

inoxx03 replied:
@gregraka Thank you for spotting this! My bad. Fixed in 6cd6d68
CC @bergerhoffer

bergerhoffer reviewed (commented):
Just one link that needs to be fixed!

@gregraka Once that link is fixed, you can approve the PR (if the update looks good to you) and you can count all of the previous approvals, including mine as a third, and merge. Thanks!

cbudz commented:
Adding my "Ack" to the chorus here.

inoxx03 commented on ADMIN.adoc:
```diff
@@ -0,0 +1,37 @@
+
+Every 2 weeks a different project member assumes the role of repository administrator.
+Signing up to be an administrator is voluntary.
+To volunteer to be an administrator, contact link:https://github.com/cbudz[@cbudz] or link:https://github.com/cbudz[@bergerhoffer].
```
```suggestion
To volunteer to be an administrator, contact link:https://github.com/cbudz[@cbudz] or link:https://github.com/bergerhoffer[@bergerhoffer].
```

gregraka reviewed (approved):
LGTM!

Let's merge this puppy!! :^)
